CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivityChains.uc

```diff
@@ -10,30 +10,408 @@ class X2StrategyElement_DefaultActivityChains extends X2StrategyElement;
 static function array<X2DataTemplate> CreateTemplates()
 {
 	local array<X2DataTemplate> Activites;
-
+	
+	Activites.AddItem(CreateCounterDarkEventTemplate());
 	Activites.AddItem(CreateSupplyRaidTemplate());
+	Activites.AddItem(CreateCaptureVIPTemplate());
+	Activites.AddItem(CreateRescueScientistTemplate());
+	Activites.AddItem(CreateRescueEngineerTemplate());
+	Activites.AddItem(CreateJailbreakFactionSoldierTemplate());
+	Activites.AddItem(CreateJailbreakCapturedSoldierTemplate());
+	Activites.AddItem(CreateGatherSuppliesTemplate());
+	Activites.AddItem(CreateGatherIntelTemplate());
+	Activites.AddItem(CreateLandedUFOTemplate());
+	//Activites.AddItem(CreateHuntChosenTemplate());
+	Activites.AddItem(CreateDestroyFacilityTemplate());
 
 	return Activites;
 }
 
-static function X2DataTemplate CreateSupplyRaidTemplate ()
+static function X2DataTemplate CreateCounterDarkEventTemplate()
 {
 	local X2ActivityChainTemplate Template;
 
-	`CREATE_X2TEMPLATE(class'X2ActivityChainTemplate', Template, 'ActivityChain_SupplyRaid');
+	`CREATE_X2TEMPLATE(class'X2ActivityChainTemplate', Template, 'ActivityChain_CounterDarkEvent');
 	
 	Template.ChooseFaction = ChooseMetFaction;
 	Template.ChooseRegions = ChooseRandomContactedRegion;
+	Template.SpawnInDeck = false;
+	
+	Template.SetupChain = SetupDarkEventChain;
+	Template.CleanupChain = CleanupDarkEventChain;
 
-	Template.Stages.AddItem('Activity_Recover');
 	Template.Stages.AddItem('Activity_PrepareCounterDE');
-	Template.Stages.AddItem('Activity_NeutralizeCommander');
-	// TODO
+	Template.Stages.AddItem('Activity_CounterDarkEvent');
+	
+	return Template;
+}
+
+static function SetupDarkEventChain(XComGameState NewGameState, XComGameState_ActivityChain ChainState)
+{
+	ChainState.PauseChainDarkEvent(NewGameState);
+}
+
+static function CleanupDarkEventChain(XComGameState NewGameState, XComGameState_ActivityChain ChainState)
+{
+	ChainState.CounterChainDarkEvent(NewGameState);
+}
+
+static function X2DataTemplate CreateSupplyRaidTemplate()
+{
+	local X2ActivityChainTemplate Template;
+
+	`CREATE_X2TEMPLATE(class'X2ActivityChainTemplate', Template, 'ActivityChain_SupplyRaid');
+	
+	Template.ChooseFaction = ChooseMetFaction;
+	Template.ChooseRegions = ChooseRandomContactedRegion;
+	Template.SpawnInDeck = true;
+	Template.NumInDeck = 1;
+	Template.DeckReq = AlwaysAvailable;
```

Comment:
This is a bit confusing - a better solution would be to make `DeckReq` return `true` by default and set a different callback only if you actually need some custom logic

Reply:
Tried that, couldn't figure it out so just did this.